File: Arcade/Intro/C/makeArrayConsecutive2.c
// Arrays are already defined with this interface:
// typedef struct arr_##name {
//   int size;
//   type *arr;
// } arr_##name;
//
// arr_##name alloc_arr_##name(int len) {
//   arr_##name a = {len, len > 0 ? malloc(sizeof(type) * len) : NULL};
//   return a;
// }
//
//
int solution(arr_integer statues) {
    int max = statues.arr[0], min = max;
    for(int i = 1; i <= statues.size - 1; i++){
        if(statues.arr[i] > max){
            max = statues.arr[i];
        }
        if(statues.arr[i] < min){
            min = statues.arr[i];
        }
    }
    return (max - min) - (statues.size - 1);
}

File: Arcade/The Core/JS/makeArrayConsecutive2.js
//SOLUTION 1:

// const solution = (statues) => {
//     return statues.sort((a, b) => a - b).reduce((result, item, index) => {
//         return result += (index == statues.length - 1) ? 0 : (statues[index + 1] - item - 1)
//     }, 0)
// }

// const solution = (statues) => statues.sort((a, b) => a - b).reduce((result, item, index) => result += (index == statues.length - 1) ? 0 : (statues[index + 1] - item - 1), 0)

//SOLUTION 2:

const solution = (statues) =>
  Math.max(...statues) - Math.min(...statues) - statues.length + 1;
